rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -213,14 +210,19 @@ private boolean allowByMinerNotBanned(Peer sender, Message message) {
     private boolean allowByMessageUniqueness(Peer sender, Message message) {
         Keccak256 encodedMessage = new Keccak256(HashUtil.keccak256(message.getEncoded()));
 
+        Set<Keccak256> receivedMessages = receivedPeerMessages.getOrDefault(sender, Collections.synchronizedSet(new HashSet<>()));
         boolean contains = receivedMessages.contains(encodedMessage);
 
         if (!contains) {
             if (message.getMessageType() == MessageType.BLOCK_MESSAGE || message.getMessageType() == MessageType.TRANSACTIONS) {
```

Comment:
A message is only added if this condition is reached, so maybe, we could check the MessageType at the beginning of the method and directly return false if the message is not a block or a transaction type.

Reply:
Addressed